cmd/snap-update-ns/common.go

```diff
@@ -0,0 +1,68 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2019 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package main
+
+import (
+	"fmt"
+
+	"github.com/snapcore/snapd/osutil"
+)
+
+type CommonProfileUpdateContext struct {
+	// instanceName is the name of the snap instance to update.
+	instanceName string
+
+	currentProfilePath string
+	desiredProfilePath string
+}
+
+func (up *CommonProfileUpdateContext) Lock() (unlock func(), err error) {
```

Comment:
should we use  upc or upCtx instead of up now?

Reply:
How about just `ctx`? It's short and it's a context :)

Reply:
not strongly against that, the question is whether that will make people think of context.Context